### PR TITLE
Load external rules config file

### PR DESCRIPTION
**Pull request by ignaciobolonio**
The rules configuration is currently a parameter in the initialisation of the Config class. This PR adds the possibility to load an external Python file as rules configuration in runtime.
It also adds the same functionality to the cli.
Example:
```
config = Config(rules=["CrossAccountTrustRule"], aws_account_id="123456789", stack_name="mockstack",)
config.load_rules_config_file(path=f"{project_location}/config/rules_config.py")
```
Example rules config file:
```
RULES_CONFIG = {
    "CrossAccountTrustRule": RuleConfig(
        filters=[
            Filter(
                rule_mode=RuleMode.WHITELISTED,
                eval={
                    "and": [
                        {"eq": [{"ref": "config.stack_name"}, "mockstack"]},
                        {"eq": [{"ref": "logical_id"}, "RootRoleOne"]},
                    ]
                },
            )
        ],
    )
}

### CODE REVIEW

File: cfripper/cli.py [hunk 1]
```diff
@@ -138,6 +145,11 @@ def process_template(
     help="Logging level",
     show_default=True,
 )
+@click.option(
+    "--rules-config-file",
+    type=click.Path(exists=True, resolve_path=True, readable=True, file_okay=False),
+    help="Loads rules configuration file",
```
Comment (ocrawford555) [inline cfripper/cli.py:151]: Maybe make the help text slightly more explicit? - `Loads rules configuration file for filtering (type: [.py, .pyc])`.

File: tests/test_files/config/rules_config_invalid.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from cfripper.config.filter import Filter
+from cfripper.config.rule_config import RuleConfig
+from cfripper.model.enums import RuleMode
+
+RULES_CONFIG = [
+    RuleConfig(
```
Comment (ocrawford555) [inline tests/test_files/config/rules_config_invalid.py:6]: I would add a comment above `RULES_CONFIG` to make it super clear why it is an invalid config (in this case missing a key)

File: cfripper/config/config.py [hunk 3]
```diff
@@ -159,3 +169,23 @@ def get_whitelisted_rules(self) -> List[str]:
...
+            if ext not in [".py", ".pyc"]:
+                raise RuntimeError("Configuration file should have a valid Python extension.")
+            spec = importlib.util.spec_from_file_location(module_name, filename)
+            module = importlib.util.module_from_spec(spec)
+            sys.modules[module_name] = module
+            spec.loader.exec_module(module)
+            rules_config = vars(module).get("RULES_CONFIG")
+            self.rules_config_schema.validate(rules_config)
+            self.rules_config = rules_config
+        except Exception:
```
Comment (ocrawford555) [inline cfripper/config/config.py:189]: Can you catch `SchemaUnexpectedTypeError` for a slightly more specific error? Will the exception itself say what the issue with the schema is? 😄

**Review by ocrawford555 — COMMENTED** [review]
Awesome stuff @ignaciobolonio .. I've left a couple of very minor comments for review.

cc @oscarbc96 should we add a CHANGELOG in this PR as well?

File: cfripper/config/config.py (hunk 3 above)
Comment (ignaciobolonio) [reply cfripper/config/config.py:189]: The logger.exception should print the information regarding that exception if I am not wrong. There is no specific action to be taken in that case. This is how the error looks like in the console:

```schema.SchemaUnexpectedTypeError: [RuleConfig(rule_mode=None, risk_value=None, filters=[Filter(reason='', eval=<function build_evaluator.<locals>.<lambda> at 0x10f972268>, rule_mode=<RuleMode.WHITELISTED: 'WHITELISTED'>, risk_value=None)])] should be instance of 'dict'```

Comment (ocrawford555) [reply cfripper/config/config.py:189]: Looks good to me, thanks for confirming 👍

**Review by ocrawford555 — DISMISSED** [review]

**Review by ocrawford555 — DISMISSED** [review]

File: cfripper/cli.py [hunk 4]
```diff
@@ -138,6 +145,11 @@ def process_template(
     help="Logging level",
     show_default=True,
 )
+@click.option(
+    "--rules-config-file",
+    type=click.File("r"),
+    help="Loads rules configuration file for filtering (type: [.py, .pyc])",
```
Comment (oscarbc96) [inline cfripper/cli.py:151]:
```suggestion
    help="Loads rules configuration file (type: [.py, .pyc])",
```
It also setups rule modes, it's not just filtering

File: cfripper/config/config.py [hunk 5]
```diff
@@ -68,6 +76,8 @@ class Config:
         "directconnect:",
     ]
 
+    rules_config_schema = Schema({str: RuleConfig})
```
Comment (oscarbc96) [inline cfripper/config/config.py:79]: What's the difference between schema and pydantic for model validation?

**Review by oscarbc96 — COMMENTED** [review]
A couple of questions.
The rest LGTM 🚀

File: cfripper/config/config.py (hunk 5 above)
Comment (ignaciobolonio) [reply cfripper/config/config.py:79]: In this case we do not need any parsing for this structure, so I assumed a pydantic model class was not needed. In order to do a simple schema validation, this was the simplest way I could think of.

**Review by ocrawford555 — APPROVED** [review]

**Review by ocrawford555 — APPROVED** [review]

**Review by oscarbc96 — APPROVED** [review]

**Pull request merged**